src/app/manifest/manifest-service-json.ts

```diff
@@ -24,6 +25,44 @@ export interface ManifestServiceState {
   statusText?: string;
 }
 
+type Mutable<T> = { -readonly [P in keyof T]: Mutable<T[P]> };
+/** Functions that can reduce the size of a table after it's downloaded but before it's saved to cache. */
+const tableTrimmers = {
+  DestinyInventoryItemDefinition(table: { [hash: number]: DestinyInventoryItemDefinition }) {
+    for (const key in table) {
+      const def = table[key] as Mutable<DestinyInventoryItemDefinition>;
+
+      // Deleting properties can actually make memory usage go up as V8 replaces some efficient
+      // structures from JSON parsing. Only replace objects with empties, and always test with the
+      // memory profiler. Don't assume that deleting something makes this smaller.
+
+      def.action = emptyObject();
+      def.backgroundColor = emptyObject();
+      def.translationBlock = emptyObject();
+      if (def.equippingBlock?.displayStrings?.length) {
+        def.equippingBlock.displayStrings = emptyArray();
+      }
+      if (def.preview?.derivedItemCategories?.length) {
+        def.preview.derivedItemCategories = emptyArray();
+      }
+      if (def.inventory.bucketTypeHash !== 3284755031) {
+        def.talentGrid = emptyObject();
+      }
+
+      if (def.sockets) {
+        def.sockets.intrinsicSockets = emptyArray();
+        for (const socket of def.sockets.socketEntries) {
+          if (socket.reusablePlugSetHash && socket.reusablePlugItems.length > 0) {
```

Comment:
is the assumption here that we can empty out `reusablePlugItems` and use `reusablePlugSetHash` to fetch the same data? maybe i am missing something but it doesn't seem like there's logic to do that.

does removing all `reusablePlugItems` entries break this? https://github.com/DestinyItemManager/DIM/blob/ed62ef58924a8c27f804ba67ba763d6737da15cb/src/app/inventory/store/sockets.ts#L225

Reply:
I think it still works - I'm using `reusablePlugSetHash` to show options in the mod viewer, but no longer populating the options on the socket itself. Generally we're using the reusable plugs from the live profile data instead of the manifest definitions, so everything displays right.

The only place we could have been using this is in Collections, but it appears we weren't making use of that there in the first place.

The main reason this is important is that the definitions currently include a list of every masterwork mod (100 entries) for every item that could be masterworked.

Reply:
Let me fix this up separately.